plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/inspections/UnusedReceiverParameterInspection.kt

```diff
@@ -113,6 +114,16 @@ class UnusedReceiverParameterInspection : AbstractKotlinInspection() {
                     RemoveReceiverFix(inSameClass)
                 )
             }
+
+            private fun KtCallableDeclaration.isCalledWithCallOrQualifiedReceiver(): Boolean {
+                return ReferencesSearch.search(this).any {
```

Comment:
I'm a bit afraid that this approach can lead us to some performance penalties... Let me think a bit more about it